renku/core/management/datasets.py

```diff
@@ -453,7 +454,7 @@ def _add_from_urls(
         destination.mkdir(parents=True, exist_ok=True)
 
         files = []
-        max_workers = min(os.cpu_count() + 4, 8)
+        max_workers = min(os.cpu_count() - 2, 4) or 1
```

Comment:
Why do you decrease the number of workers? These are values recommended by Python and worked fine so far. Also, `os.cpu_count() - 2` can be negative when there is one core.

Reply:
Where is this recommendation coming from? I reduce it, since the wait has to be equal to the thread pool size due to the throttling.

Reply:
I got it from https://docs.python.org/3/library/concurrent.futures.html and then just reduced the max value. Basically, since downloading is an IO-bound operation, Python library used 5x more download thread up to 3.5 and they adjust it to be a bit lower from 3.6 upwards.
I believe UX will be a bit degraded with these values but we can adjust later if it is the case.